include/ginkgo/core/preconditioner/parse_limit.hpp

```diff
@@ -0,0 +1,37 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef GKO_PUBLIC_CORE_PRECONDITIONER_PARSE_LIMIT_HPP_
+#define GKO_PUBLIC_CORE_PRECONDITIONER_PARSE_LIMIT_HPP_
+
+
+#include <type_traits>
+
+
+#include <ginkgo/core/preconditioner/isai.hpp>
+
+
+namespace gko {
+namespace preconditioner {
+namespace detail {
+
+
+// true_type if ExplicitType is an instantiation of TemplateType.
+template <typename ExplicitType, template <typename...> class TemplateType>
+struct is_instance_of : std::false_type {};
```

Comment:
IMO `is_instantiation_of` would be a better name (it also matches the docs better)